### PR TITLE
build(nix): Introduce flake.formatter for `nix fmt`

### PR DESCRIPTION
**Pull request by ditsuke**
## What and why
Introduces a `formatter` in the repo's Nix flake. This enables the usage of `nix fmt` to format the Nix files in the repo reproducibly.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -131,6 +135,11 @@
             ...
           }:
           {
+            # For standardised reproducible formatting with `nix fmt`
+            # HELP: why does the per system formatter not work?
+            # formatter = inputs.nixfmt.packages.${system}.nixfmt;
```
Comment (ditsuke) [inline flake.nix:140]: This doesn't work, and I couldn't quite figure out why. From the error message it appears to try building the aarch_64 darwin build in x86_64-linux.

```
       error: a 'aarch64-darwin' with features {} is required to build '/nix/store/apv47ywrf96wyiy5cd9l9grz0kmg1y1y-cabal2nix-nixfmt.drv', but I am a 'x86_64-linux' with features {benchmark,
 big-parallel, kvm, nixos-test, uid-range}
```

Comment (ditsuke) [reply flake.nix:140]: Oh, I see. Do you see a solution that does not require changes in `nixfmt`'s flake? Else maybe we can use nixfmt from nixpkgs, plus point being we get the builds cached for free.

**Comment by leana8959** [comment]
What about [alejandra](https://github.com/kamadorueda/alejandra) as the nix formatter? It opts for "tall" layouts, which helps git diff.

**Comment by ditsuke** [comment]
> What about [alejandra](https://github.com/kamadorueda/alejandra) as the nix formatter? It opts for "tall" layouts, which helps git diff.

I quite like alejandra's style, personally. This repo has been already using nixfmt's rfc101 branch, but if the consensus is with a switch to alejandra we could do that here.

**Comment by SomeoneSerge** [comment]
> What about [alejandra](https://github.com/kamadorueda/alejandra) as the nix formatter? It opts for "tall" layouts, which helps git diff.

I think there are many similarities between alejandra and rfc101, and both aim to optimize git diffs. Personally, I dislike them both, and we only had chosen the experimental nixfmt in @philiptaron's rewrite because it has a chance of wider adoption (nixpkgs)

File: flake.nix (hunk 1 above)
Comment (SomeoneSerge) [reply flake.nix:140]: (sorry, can't suggest anything rn, but can review your proposals; I'm more available next week)

**Review by philiptaron — CHANGES_REQUESTED** [review]
You can use the [RFC 166](https://github.com/NixOS/rfcs/pull/166) by pulling it from nixpkgs, rather than the flake. See [this discourse post](https://discourse.nixos.org/t/call-for-testing-nix-formatter/39179) from @piegamesde. 

The name of the package is `nixfmt-rfc-style`.

As this PR notes, the flake.nix in [piegamesde/nixfmt/rfc101-style](https://github.com/piegamesde/nixfmt/tree/rfc101-style) has multiple issues. Let's side-step those by relying on nixpkgs.

**Comment by ditsuke** [comment]
> You can use the [RFC 166](https://github.com/NixOS/rfcs/pull/166) by pulling it from nixpkgs, rather than the flake. See [this discourse post](https://discourse.nixos.org/t/call-for-testing-nix-formatter/39179) from @piegamesde.
> 
> The name of the package is `nixfmt-rfc-style`.
> 
> As this PR notes, the flake.nix in [piegamesde/nixfmt/rfc101-style](https://github.com/piegamesde/nixfmt/tree/rfc101-style) has multiple issues. Let's side-step those by relying on nixpkgs.

Thank you, pulling in from nixpkgs is so much better.

**Review by philiptaron — APPROVED** [review]
Looks good. I intend to merge when CI passes.

**Pull request merged**